tests/substrate_tests/calls.rs

```diff
@@ -216,6 +216,32 @@ fn try_catch_external_calls() {
     runtime.constructor(0, Vec::new());
     runtime.function_expect_failure("test", Vec::new());
 
+    let mut runtime = build_solidity(
+        r##"
+        contract c {
+            function test() public {
+                other o = new other();
+                int32 x = 0;
+                try o.test() {
+                    x = 1;
+                } catch (bytes c) {
+                    assert(c == hex"a079c3080c666f6f");
+                    x = 2;
+                }
+                assert(x == 2);
+            }
+        }
+
+        contract other {
+            function test() public returns (int32, bool) {
+                revert("foo");
+            }
+        }
+        "##,
+    );
+
+    runtime.constructor(0, Vec::new());
+    runtime.function_expect_failure("test", Vec::new());
```

Comment:
I think it is worth adding a case in which we enter the `catch` branch and the contract does not fail.

Reply:
You're right, that was a mistake. I've fixed it.